src/main/java/com/sparrowwallet/drongo/crypto/ECKey.java

```diff
@@ -488,53 +488,36 @@ public boolean verify(Sha256Hash sigHash, TransactionSignature signature) {
     }
 
     public ECKey getTweakedOutputKey() {
-        TaprootPubKey taprootPubKey = liftX(getPubKeyXCoord());
-        ECPoint internalKey = taprootPubKey.ecPoint;
-        byte[] taggedHash = Utils.taggedHash("TapTweak", internalKey.getXCoord().getEncoded());
-        ECKey tweakValue = ECKey.fromPrivate(taggedHash);
-        ECPoint outputKey = internalKey.add(tweakValue.getPubKeyPoint());
-
-        if(hasPrivKey()) {
-            BigInteger taprootPriv = priv;
-            BigInteger tweakedPrivKey = taprootPriv.add(tweakValue.getPrivKey()).mod(CURVE_PARAMS.getCurve().getOrder());
-            //TODO: Improve on this hack. How do we know whether to negate the private key before tweaking it?
-            if(!ECKey.fromPrivate(tweakedPrivKey).getPubKeyPoint().equals(outputKey)) {
-                taprootPriv = CURVE_PARAMS.getCurve().getOrder().subtract(priv);
-                tweakedPrivKey = taprootPriv.add(tweakValue.getPrivKey()).mod(CURVE_PARAMS.getCurve().getOrder());
-            }
+        BigInteger curveOrderN = CURVE_PARAMS.getCurve().getOrder();
+        ECPoint pubKey = pub.get();  // our internal key’s *actual* public key (before normalization to even y)
 
-            return new ECKey(tweakedPrivKey, outputKey, true);
+        boolean yWasOdd = pubKey.getAffineYCoord().toBigInteger().testBit(0);
```

Comment:
Rather add a method to `LazyECPoint` called `isOddYCoord()` with the same test. I like this test - it is efficient and safe because `decodePoint()` always returns points in normalized affine form since it's decoding from standard compressed (33  bytes) or uncompressed (65 bytes) point encoding.